okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelectorImpl.java

```diff
@@ -98,7 +98,9 @@ private static String switchWsToHttp(String url) {
 
     @Override
     public Optional<HttpUrl> redirectToNext(HttpUrl current) {
-        int index = currentUrl.updateAndGet(operand -> (operand + 1) % baseUrls.size());
+        Optional<Integer> currentIndex = indexFor(current);
```

Comment:
can you leave a comment indicating what the purpose of this logic is?

Reply:
added a comment, not entirely sure how clear it is
(also renamed the argument to be a little less confusing - url argument named `current` vs int field named `currentUrl`)